java/org/apache/catalina/webresources/AbstractSingleArchiveResourceSet.java

```diff
@@ -91,6 +113,9 @@ public AbstractSingleArchiveResourceSet(WebResourceRoot root, String webAppMount
 
     @Override
     protected JarEntry getArchiveEntry(String pathInArchive) {
+        if (isCacheArchiveEntries()) {
+            return archiveEntries.get(pathInArchive);
```

Comment:
I wonder if we need to reproduce the any of logic being done in ZipFile.getEntry() (which JarFile.getEntry() delegates to), specifically the add slash logic
```
 public ZipEntry getEntry(String name) {
        if (name == null) {
            throw new NullPointerException("name");
        }
        long jzentry = 0;
        synchronized (this) {
            ensureOpen();
            jzentry = getEntry(jzfile, zc.getBytes(name), true);
            if (jzentry != 0) {
                // If no entry is found for the specified 'name' and
                // the 'name' does not end with a forward slash '/',
                // the implementation tries to find the entry with a
                // slash '/' appended to the end of the 'name', before
                // returning null. When such entry is found, the name
                // that actually is found (with a slash '/' attached)
                // is used
                // (disabled if jdk.util.zip.ensureTrailingSlash=false)
                ZipEntry ze = ensuretrailingslash ? getZipEntry(null, jzentry)
                                                  : getZipEntry(name, jzentry);
                freeEntry(jzfile, jzentry);
                return ze;
            }
        }
        return null;
    }
```

Reply:
I think they're the same entries...we're using `JarFile.entries()` to load the map, which gives us a `JarEntryIterator` that's returning `ZipEntry` instances

Reply:
right, im just worried that the lookup will be different
for example:

previously i think a lookup for a name of `myresource` would return a value which had a entry.getName() of `myresource/` but our hashmap is doing a strict string lookup

Reply:
I think I'm leaning toward backing this change out...it wasn't involved in the initial problem identified in the ticket, I just added it here because I thought it was an opportunity for additional optimization.  CI says we don't appear to be exposed to slash issues currently, but that could certainly change in the future as the archives in the application change.  I'd rather remove the exposure to side effects because that code snippet you posted is rather hairy.

https://bugs.openjdk.java.net/browse/JDK-8185136